code/game/turfs/open/_open.dm

```diff
@@ -270,3 +270,8 @@
 		air.assert_gas(/datum/gas/pluoxium)
 		air.gases[/datum/gas/pluoxium][MOLES]+=(pulse_strength/4000)
 		air.garbage_collect()
+	if (air.gases[/datum/gas/hydrogen])
+		pulse_strength = min(pulse_strength, air.gases[/datum/gas/hydrogen][MOLES] * 1000)
+		air.gases[/datum/gas/hydrogen][MOLES] = max(air.gases[/datum/gas/hydrogen][MOLES] - (pulse_strength/1000), 0)
+		air.assert_gas(/datum/gas/tritium)
+		air.gases[/datum/gas/tritium][MOLES] += (pulse_strength / 1000)
```

Comment:
move the air.garbage_collect() line below both the if statements, or put it in both. It handles negative molar counts and all that hellish stuff, if you remove gas outside of react code it'll need to garbage collect. Also might wanna throw an `air_update_turf()` here too, pretty sure both if statements need it